packages/lang-server/test/service_url.test.ts

```diff
@@ -0,0 +1,134 @@
+  })
+
+  it('when ref is undefined should return undefined', async () => {
+    expect(await getServiceUrl(editorWorkspace, ctx)).toBeUndefined()
+  })
+
+  it('when element does not have service url annotation should return undefined', async () => {
+    ctx.ref = {
+      element: new ObjectType({ elemID: new ElemID('salesforce', 'Account') }),
+      path: [],
+      isList: false,
+    }
+
+    workspace.getValue = mockFunction<Workspace['getValue']>().mockResolvedValue(
+      new ObjectType({ elemID: new ElemID('salesforce', 'Account') })
+    )
+
+    expect(await getServiceUrl(editorWorkspace, ctx)).toBeUndefined()
+    expect(workspace.getValue).toHaveBeenCalledWith(ctx.ref.element.elemID)
+  })
+
+  it('when getValue return non-element value should return undefined', async () => {
+    ctx.ref = {
+      element: new ObjectType({ elemID: new ElemID('salesforce', 'Account') }),
+      path: [],
+      isList: false,
+    }
+
+    workspace.getValue = mockFunction<Workspace['getValue']>().mockResolvedValue('aaa')
+
+    expect(await getServiceUrl(editorWorkspace, ctx)).toBeUndefined()
+    expect(workspace.getValue).toHaveBeenCalledWith(ctx.ref.element.elemID)
+  })
+
+  it('should return the url of the element in the context', async () => {
+    ctx.ref = {
+      element: new ObjectType({ elemID: new ElemID('salesforce', 'Account') }),
+      path: [],
+      isList: false,
+    }
+
+    workspace.getValue = mockFunction<Workspace['getValue']>().mockResolvedValue(
+      new ObjectType({ elemID: new ElemID('salesforce', 'Account'), annotations: { [CORE_ANNOTATIONS.SERVICE_URL]: 'https://someurl.com' } })
+    )
+
+    expect(await getServiceUrl(editorWorkspace, ctx)).toEqual(new URL('https://someurl.com'))
+    expect(workspace.getValue).toHaveBeenCalledWith(ctx.ref.element.elemID)
+  })
+
+  it('when parent has annoation and child does not should return the url of the parent', async () => {
+    const type = new ObjectType({ elemID: new ElemID('salesforce', 'Account') })
+    const field = new Field(type, 'fieldName', BuiltinTypes.NUMBER)
+    type.fields[field.name] = field
+
+    ctx.ref = {
+      element: field,
+      path: [],
+      isList: false,
+    }
+
+    const typeWithAnnotations = type.clone()
+    typeWithAnnotations.annotations = { [CORE_ANNOTATIONS.SERVICE_URL]: 'https://someurl.com' }
+
+    workspace.getValue = mockFunction<Workspace['getValue']>().mockResolvedValue(
+      typeWithAnnotations
+    )
+
+    expect(await getServiceUrl(editorWorkspace, ctx)).toEqual(new URL('https://someurl.com'))
+    expect(workspace.getValue).toHaveBeenCalledWith(type.elemID)
+  })
+
+  it('when parent has annoation and child has annotation return the url of the child', async () => {
+    const type = new ObjectType({ elemID: new ElemID('salesforce', 'Account') })
+    const field = new Field(type, 'fieldName', BuiltinTypes.NUMBER)
+    type.fields[field.name] = field
```

Comment:
style nit: we should not create fields like this. fields are coupled to the object type, they should be created through the object type constructor
```suggestion
    const type = new ObjectType({
      elemID: new ElemID('salesforce', 'Account'),
      fields: {
        fieldName: { type: BuiltinTypes.NUMBER },
      }
    })
    const field = type.fields.fieldName
```